Is RefactorLog supported?

The .sqlproj has support for RefactorLog. Somehow when your rename columns in the visual editor it generates this file which is also included in the project as:
```xml
 <ItemGroup>
    <RefactorLog Include="CallProSQL.refactorlog" />
  </ItemGroup>
```
But I cannot add this to the SDK style project.  

So I guess this means there is no native support for handling column renames in the SDK style project? I should do it myself in a pre-deployment step? and craft the sp_rename myself, but then I also have to handle whether it is needed, that seems boilerplate?

P.S. I always found this RefactorLog file a bit scary but it works very well, and seems to do something similar (but simpler) as EF MigrationHistory.

So I don't think I understand that sample. My project looks like this:
```xml
<Project Sdk="MSBuild.Sdk.SqlProj/2.8.0">
    <PropertyGroup>
        <TargetFramework>netstandard2.0</TargetFramework>
        <SqlServerVersion>Sql150</SqlServerVersion>
        <RunSqlCodeAnalysis>false</RunSqlCodeAnalysis>
        <!-- For additional properties that can be set here, please refer to https://github.com/rr-wfm/MSBuild.Sdk.SqlProj#model-properties -->
    </PropertyGroup>

    <PropertyGroup>
        <!-- Refer to https://github.com/rr-wfm/MSBuild.Sdk.SqlProj#publishing-support for supported publishing options -->
    </PropertyGroup>
	
	<ItemGroup>
      <Content Remove="Scripts/**/*.sql" />
      <Content Remove="CallPro.refactorlog" />
    </ItemGroup>

	<ItemGroup>
		<PostDeploy Include="Scripts\Post-Deployment\Script.PostDeployment.sql" />
		<PreDeploy Include="Scripts\Script.PreDeployment.sql" />
		<RefactorLog Include="CallPro.refactorlog" />
	</ItemGroup>
	
    <ItemGroup>
      <Folder Include="bin\Debug\netstandard2.0\" />
    </ItemGroup>

</Project>
```

I pasted the ItemGroupd from the test folder but when I compile now it gives me:

```
1>Writing model to C:\data\devops\calway\callpro\callpro\Database\CallPro\obj\Debug\netstandard2.0\CallPro.dacpac
1>Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
1> ---> Microsoft.SqlServer.Dac.DacServicesException: Could not save package to file.
1>Could not add refactor log file 'C:\data\devops\calway\callpro\callpro\Database\CallPro\CallPro.refactorlog' to package.
1>Data at the root level is invalid. Line 1, position 1.
1> ---> Microsoft.Data.Tools.Schema.Sql.Build.SqlPackageException: Could not add refactor log file 'C:\data\devops\calway\callpro\callpro\Database\CallPro\CallPro.refactorlog' to package.
1>Data at the root level is invalid. Line 1, position 1.
1> ---> System.Xml.XmlException: Data at the root level is invalid. Line 1, position 1.
1>   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
1>   at System.Xml.XmlTextReaderImpl.Throw(String res, String arg)
1>   at System.Xml.XmlTextReaderImpl.ParseRootLevelWhitespace()
1>   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
1>   at System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean preserveWhitespace)
1>   at System.Xml.XmlDocument.Load(XmlReader reader)
1>   at Microsoft.Data.Tools.Schema.Common.XmlUtilities.LoadWithProhibitDtd(XmlDocument doc, Stream stream)
1>   at Microsoft.Data.Tools.Schema.Sql.Refactoring.SqlRefactoringLog.ValidateRefactoringLog(Stream refactorLogStream, String refactorLogPathForErrorMessage)
1>   at Microsoft.Data.Tools.Schema.Sql.Build.DacUtilities.AddRefactorLogToSqlPackage(SqlPackage package, Stream inputLogStream, String refactorLogPath)
1>   at Microsoft.Data.Tools.Schema.Sql.Build.DacUtilities.AddRefactorLogToSqlPackage(SqlPackage package, String refactorLogPath)
1>   --- End of inner exception stack trace ---
1>   at Microsoft.Data.Tools.Schema.Sql.Build.DacUtilities.AddRefactorLogToSqlPackage(SqlPackage package, String refactorLogPath)
1>   at Microsoft.SqlServer.Dac.Model.SqlSchemaModelObjectService.<>c__DisplayClass77_0.<BuildPackage>b__0()
1>   at Microsoft.SqlServer.Dac.Model.SqlSchemaModelObjectService.ExecutePackageOperationWithErrorHandling(Action dacPackageOperation)
1>   --- End of inner exception stack trace ---
1>   at Microsoft.SqlServer.Dac.Model.SqlSchemaModelObjectService.ExecutePackageOperationWithErrorHandling(Action dacPackageOperation)
1>   at Microsoft.SqlServer.Dac.Model.SqlSchemaModelObjectService.BuildPackage(Func`1 streamGetter, PackageMetadata packageMetadata, PackageOptions packageOptions)
1>   at Microsoft.SqlServer.Dac.Model.TSqlModel.BuildPackage(String packageFilePath, PackageMetadata packageMetadata, PackageOptions packageOptions)
1>   at Microsoft.SqlServer.Dac.DacPackageExtensions.BuildPackage(String packageFilePath, TSqlModel model, PackageMetadata packageMetadata, PackageOptions packageOptions)
1>   at MSBuild.Sdk.SqlProj.DacpacTool.PackageBuilder.SaveToDisk(FileInfo outputFile, PackageOptions packageOptions) in /home/runner/work/MSBuild.Sdk.SqlProj/MSBuild.Sdk.SqlProj/src/DacpacTool/PackageBuilder.cs:line 165
1>   at MSBuild.Sdk.SqlProj.DacpacTool.Program.BuildDacpac(BuildOptions options) in /home/runner/work/MSBuild.Sdk.SqlProj/MSBuild.Sdk.SqlProj/src/DacpacTool/Program.cs:line 185
1>   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
1>   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
1>   --- End of inner exception stack trace ---
1>   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
1>   at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
1>   at System.Delegate.DynamicInvokeImpl(Object[] args)
1>   at System.CommandLine.NamingConventionBinder.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
1>   at System.CommandLine.Parsing.ParseResultExtensions.InvokeAsync(ParseResult parseResult, IConsole console)
1>   at MSBuild.Sdk.SqlProj.DacpacTool.Program.Main(String[] args) in /home/runner/work/MSBuild.Sdk.SqlProj/MSBuild.Sdk.SqlProj/src/DacpacTool/Program.cs:line 80
1>   at MSBuild.Sdk.SqlProj.DacpacTool.Program.<Main>(String[] args)
1>c:\data\nuget\packages\msbuild.sdk.sqlproj\2.8.0\Sdk\Sdk.targets(243,5): error MSB3073: The command "dotnet "c:\data\nuget\packages\msbuild.sdk.sqlproj\2.8.0\Sdk\../tools/net8.0/DacpacTool.dll" build -o "obj\Debug\netstandard2.0\CallPro.dacpac" -n "CallPro" -v "1.0.0" -sv Sql150 -i "obj\Debug\netstandard2.0\CallPro.InputFiles.txt"    -dp IncludeCompositeObjects=true --predeploy Scripts\Script.PreDeployment.sql --postdeploy Scripts\Post-Deployment\Script.PostDeployment.sql --refactorlog CallPro.refactorlog        " exited with code -532462766.
1>Done building project "CallPro.csproj" -- FAILED.
```
![devenv_hxpyVYFPSB](https://github.com/user-attachments/assets/de871eaf-7fe7-405b-823e-f687d96abe6c)

--- a/README.md
+++ b/README.md
@@ -583,7 +583,8 @@ EXEC (@Query)
 
 ## Reference `MSBuild.Sdk.SqlProj` from class library
 The output of `MSBuild.Sdk.SqlProj` is not an assembly, but a `.dacpac`. In order to correctly reference a `MSBuild.Sdk.SqlProj` based project from a class library, the `ReferenceOutputAssembly` hint needs to be set to `False`:
-```
+
+```xml
 <ItemGroup>
     <ProjectReference
       Include="../MyDacpacProj/MyDacpacProj.csproj"
@@ -591,6 +592,16 @@ The output of `MSBuild.Sdk.SqlProj` is not an assembly, but a `.dacpac`. In orde
 </ItemGroup>
 ```
 Now, upon compilation of the class library, the relevant `.dacpac` files get copied to the output directory.
+
+## Refactor Log support
+
+While the SDK does not help you maintain a [refactor log](https://learn.microsoft.com/sql/ssdt/how-to-use-rename-and-refactoring-to-make-changes-to-your-database-objects), you can use an existing one during build by referring to it in your project:
+
+```xml
+<ItemGroup>
+    <RefactorLog Include="RefactorLog\TestProjectWithPrePost.refactorlog" />
+</ItemGroup>
+```
 
 ## Known limitations
 Since this is not an entire project system but only an MSBuild SDK we cannot provide IntelliSense for objects defined within the project. This limitation can be circumvented by connecting the SQL editor to a live database that is used for development purposes.

